spec/PeriodicBackgroundSync-index.bs

```diff
@@ -157,3 +159,51 @@ The <code><dfn method for=PeriodicSyncManager title="getTags()">getTags()</dfn><
+    readonly attribute DOMString tag;
+  };
+</script>
+
+The {{PeriodicSyncEvent}} interface represents a [=firing=] [=periodicsync registration=].
+
+### Firing periodicSync events ### {#firing-periodicsync-events}
+Whenever the user agent changes to [=online=], it SHOULD [=fire a periodicSync event=] for each [=periodicsync registration=] whose [=registration state=] is [=pending=] and [=time to fire=] is now or in the past. The events MAY be fired in any order.
+
+### [=Fire a periodicSync event=] ### {#firing-a-periodicsync-event}
+The user agent will fire a [=periodicSync event=] for a [=periodicsync registration=] as soon as network connectivity is available, at or after the [=time to fire=] of the [=periodicsync registration=].
+If a [=periodicSync event=] fails, the user agent MAY <dfn>retry</dfn> it one or more times at a time of its choosing, based on some user agent defined heuristics. The interval between [=retry=] attempts SHOULD be much smaller than the [=minInterval=] value associated with the [=periodicsync registration=].
+
+A user agent MAY impose a time limit on the lifetime extension and execution time of a {{PeriodicSyncEvent}} which is stricter than the time limit imposed for {{ExtendableEvent}}s in general. In particular, any retries of the {{PeriodicSyncEvent}} MAY have a significantly shortened time limit.
+
+To <dfn>fire a periodicSync event</dfn> for a [=periodicsync registration=] <var>registration</var>, the user agent MUST run the following steps:
+1. [=Assert=]: |registration|'s [=registration state=] is [=pending=].
+1. Let <var>serviceWorkerRegistration</var> be the [=service worker registration=] associated with |registration|.
+1. [=Assert=]: |registration| exists in the [=list of periodicSync registrations=] associated with |serviceWorkerRegistration|.
+1. [=Assert=]: |registration|'s [=time to fire=] is equal to the current time or in the past.
+1. Set |registration|'s [=registration state=] to [=firing=].
+1. [=Fire functional event=] "<code>periodicSync</code>" using {{PeriodicSyncEvent}} on |serviceWorkerRegistration| with {{PeriodicSyncEvent/tag}} set to the [=tag=] associated with |registration|.
+1. Then Run the following steps with <var>dispatchedEvent</var>:
+    1. Let <var>waitUntilPromise</var> be the result of [=waiting for all=] of |dispatchedEvent|'s [=extend lifetime promises=].
+    1. [=Upon fulfillment=] of |waitUntilPromise|, perform the following steps atomically:
+        1. Set |registration|'s state to [=pending=].
+        1. Calculate an appropriate <var>timeToFire</var> for |registration| based on the [=minInterval=] value of its associated options, and any other considerations the user agent has implemented per recommendations in [[#privacy]] and Resource Usage. The time interval between now and |timeToFire| MUST be greater than or equal to the [=minInterval=] specified in |options|. Set |registration|'s [=time to fire=] to |timeToFire|.
```

Comment:
As mentioned previously & below: abstract this away into a section + definition of its own.

Reply:
done.